scheduler/src/cook/rest/api.clj

```diff
@@ -920,13 +918,19 @@
                 (assoc :partitions (into #{} (walk/stringify-keys partitions))))))
        (into #{})))
 
-(defn validate-job-gpu-model
-  "Validates that GPU model entered is supported on the pool"
-  [pool-name {:keys [env]}]
-  (let [requested-gpu-model (get env "COOK_GPU_MODEL")]
+(defn validate-gpu-job
+  "Validates that a job requesting GPUs is supported on the pool"
+  [gpu-enabled? pool-name {:keys [gpus env]}]
+  (let [requested-gpu-model (get env "COOK_GPU_MODEL")
+        gpus' (or gpus 0)]
+    (when (and (pos? gpus') (not gpu-enabled?))
+      (throw (ex-info (str "GPU support is not enabled") {})))
     (when (and requested-gpu-model
                (not (contains? (get-gpu-models-on-pool (config/valid-gpu-models) pool-name) requested-gpu-model)))
-      (throw (ex-info (str "The following GPU model is not supported: " requested-gpu-model) {})))))
+      (throw (ex-info (str "The following GPU model is not supported: " requested-gpu-model) {})))
+    (when (and (and (pos? gpus') (not requested-gpu-model))
+               (not (get-gpu-models-on-pool (config/valid-gpu-models) pool-name)))
+      (throw (ex-info (str "Job requested GPUs but pool " pool-name " does not have any valid GPU models") {})))))
```

Comment:
Should we add a case to the integration tests for this?

Reply:
added 2 cases:
gpus are requested and gpu model is not specified, and pool does not have default model --> submission failure
gpus are requested and gpu model is not specified, and pool does have default model --> submission success

Reply:
added case to integration tests